src/harness/testcases/WINNF_FT_S_HBT_testcase.py

```diff
@@ -0,0 +1,155 @@
+#    Copyright 2016 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+import json
+import os
+import sas
+from  util import winnforum_testcase, makePpaAndPalRecordsConsistent
+import sas_testcase
+from datetime import datetime, timedelta
+import time
+
+
+class HeartbeatTestcase(sas_testcase.SasTestCase):
+  def setUp(self):
+    self._sas, self._sas_admin = sas.GetTestingSas()
+    self._sas_admin.Reset()
+
+  def tearDown(self):
+    pass
+
+  def getHeartBeatRequest(self,cbsd_ids, grant_ids, previousTransmitExpireTime):     
+    heartbeat_request = []
+    for cbsd_id, grant_id, transmitExpireTime in zip(cbsd_ids, grant_ids, previousTransmitExpireTime):
+        status = 'AUTHORIZED'
+        if(transmitExpireTime == None):
+            status = 'GRANTED'
+        elif datetime.strptime(transmitExpireTime,'%Y-%m-%dT%H:%M:%SZ') <=  (datetime.utcnow() + timedelta(seconds= 2)):
+            time.sleep(2)
+            status = 'GRANTED'
+        heartbeat_request.append({
+            'cbsdId': cbsd_id,
+            'grantId': grant_id,
+            'operationState': status
+        })
+    return {'heartbeatRequest': heartbeat_request}
+
+  def sendAndAssertSuccessfulHeartBeat(self, cbsd_ids, grant_ids, previousTransmitExpireTime):    
+    request =  self.getHeartBeatRequest(cbsd_ids, grant_ids, previousTransmitExpireTime)
+    response = self._sas.Heartbeat(request)['heartbeatResponse']
+    # check Heartbeat response 
+    self.assertEqual(len(response), 3)
+    for response_num, resp in enumerate(response):
+        self.assertEqual(resp['cbsdId'], cbsd_ids[response_num])
+        self.assertEqual(resp['grantId'], grant_ids[response_num])
+        self.assertEqual(resp['response']['responseCode'], 0)
+        grant_expire_time_1 = datetime.strptime(resp['grantExpireTime'],'%Y-%m-%dT%H:%M:%SZ')
+        transmit_expire_time_1 = datetime.strptime(resp['transmitExpireTime'],'%Y-%m-%dT%H:%M:%SZ')
+        self.assertLessEqual((transmit_expire_time_1 - datetime.utcnow()).total_seconds(), 240)
+        self.assertLessEqual(transmit_expire_time_1, grant_expire_time_1)
+    del request
```

Comment:
There's no value to deleting the variable just before it goes out of scope anyway.

Reply:
you are right, thank you, it is done.